File: Fundamentals/LoginApp/app/src/main/java/com/example/loginapp/MainActivity.java
package com.example.loginapp;

import androidx.appcompat.app.AppCompatActivity;

import android.content.Intent;
import android.content.SharedPreferences;
import android.os.Bundle;
import android.os.Handler;

public class MainActivity extends AppCompatActivity {

    Intent intent;
    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);





        new Handler().postDelayed(new Runnable() {
            @Override
            public void run() {
                SharedPreferences pref = getSharedPreferences("sf_login", MODE_PRIVATE);
                boolean isLoggedIn = pref.getBoolean("flag", false);
                intent = new Intent(MainActivity.this, isLoggedIn ? HomeScreen.class : LoginActivity.class);
                startActivity(intent);
                finish();
            }
        }, 3000);
    }
}

File: Dependency_Injection/Dependency_Injection_Fundamentals/app/src/main/java/com/example/diintroduction/MainActivity.kt
package com.example.diintroduction

import androidx.appcompat.app.AppCompatActivity
import android.os.Bundle

class MainActivity : AppCompatActivity() {
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)

        val battery = Battery()
        val memoryCard =MemoryCard()
        val serviceProvider = ServiceProvider()
        val simCard = SimCard(serviceProvider)
        val smartPhone = SmartPhone(memoryCard, simCard)

        smartPhone.makeCallRecording()
    }
}